test/integration-tests/Seaport.test.ts

```diff
@@ -26,6 +28,69 @@ describe('Seaport', () => {
   let permit2: Permit2
   let planner: RoutePlanner
   let cryptoCovens: ERC721
+  let weth: Contract
+
+  // @dev We have to separate the ERC20 tests because we use orders taken from a much
+  // newer block. If we were to replace all of the example orders with newer ones,
+  // many existing tests would break.
+  describe('Seaport ERC20', () => {
+    let bob: SignerWithAddress
+    beforeEach(async () => {
+      await resetFork(16635782)
+      // alice's permits fail w/ account not found so using bob from default signers
+      bob = (await ethers.getSigners())[1]
+      permit2 = (await deployPermit2()).connect(bob) as Permit2
+      router = (await deployUniversalRouter(permit2)).connect(bob) as UniversalRouter
+      planner = new RoutePlanner()
+      cryptoCovens = COVEN_721.connect(bob) as ERC721
+      weth = new ethers.Contract(WETH.address, ERC20_ABI, bob)
+
+      // bob deposits 10 eth into weth
+      await bob.sendTransaction({ to: weth.address, value: ethers.utils.parseEther('10') })
+      // approve permit2 for all for bob's weth
+      await weth.connect(bob).approve(permit2.address, ethers.constants.MaxUint256)
+    })
+
+    it('completes an advanced order offering ERC20', async () => {
```

Comment:
could you add another test where `APPROVE_ERC20` doesnt need to be included because someone else has already done it? i.e. like the test does an APPROVE tx, and then does this same tx but without the APPROVE?